pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -216,6 +219,77 @@ func (m *kubeGenericRuntimeManager) getKubeletContainers(allContainers bool) ([]
 	return result, nil
 }
 
+// getKubeletContainerStatuses gets all containers' status for the pod sandbox.
+func (m *kubeGenericRuntimeManager) getKubeletContainerStatuses(podSandboxID string) ([]*kubecontainer.ContainerStatus, error) {
+	containers, err := m.runtimeService.ListContainers(&runtimeApi.ContainerFilter{
+		PodSandboxId: &podSandboxID,
+	})
+	if err != nil {
+		glog.Errorf("ListContainers error: %v", err)
+		return nil, err
+	}
+
+	statuses := make([]*kubecontainer.ContainerStatus, len(containers))
+	// TODO: optimization: set maximum number of containers per container name to examine.
+	for i, c := range containers {
+		if !isContainerManagedByKubelet(c.GetName()) {
```

Comment:
I'm not sure we need this check. If we assume the `ContainerFilter` above is accurate, then this check shouldn't matter. What are we guarding against?